### DISCUSSION TITLE
yarn failed

### DISCUSSION THREAD

**Discussion by sproutlxq** [RocketChat/Rocket.Chat#31291] [General]
Has anyone else failed to execute yarn install? There are many packages that fail to install, such as the apn and http2 packages. I don't have any problems with my network, and installing with npm is successful. But if I install with npm, executing the script npm run dev afterward doesn't start the project correctly, and in general there are a lot of inconsistencies in the dependencies that cause problems as well.

Translated with www.DeepL.com/Translator (free version)

**Comment by rezerov** [comment]
Hello @sproutlxq, which platform are you on? If it's Linux then I suggest running `yarn add deasync`and then trying to run 'yarn' again followed by 'yarn dsv'. 

I encountered a similar issue while setting up the project locally on Debian 12 (Plasma). 

Additionally, the project is set up to support yarn so using npm is not recommended. Hope this helps :)

**Reply by sproutlxq** [reply]
Thanks for your answer. I am starting local development on windows and now I can yarn correctly after switching the code from master branch to develop branch.

**Reply by rezerov** [reply]
@sproutlxq That's great to hear!

**Reply by PrathameshSonje** [reply]
@sproutlxq Hi! which version of node are you on? I am too getting multiple failures after running yarn.

**Reply by wes-q** [reply]
Hello, i'm new here, I'm trying to setup Rocketchat to develop. I've been following the instructions on the README file.

I am getting this error when running "yarn" command when using node v14
SyntaxError: Unexpected token '??='

When using node v20, the error doesnt come up but yarn tells I should be using v14.
My meteor version is 2.13.3
I tried using corepack enable but im getting same unexpected token error.
I am using volta pin command to switch node versions.

**Reply by sproutlxq** [reply]
The node version I'm using is v14.21.3

**Reply by Dnouv** [reply]
It would be good to follow the development setup instructions from the [Developer Docs Link](https://developer.rocket.chat/open-source-projects/server/server-environment-setup). It is updated more frequently than the README. 

Thank you!
